scrapers/al/bills.py

```diff
@@ -60,70 +25,133 @@ def scrape(self, session):
-                primary=True,
+                if row["Subject"]:
+                    bill.add_subject(row["Subject"].strip())
+
+                if row["CompanionInstrumentNbr"] != "":
+                    self.warning("AL Companion found. Code it up.")
+
+                # TODO: EffectiveDateCertain, EffectiveDateOther
+
+                # TODO: Fiscal notes, BUDGET ISOLATION RESOLUTION
+
+                bill.extras["AL_BILL_ID"] = row["ID"]
+
+                yield bill
+
+            # no need to paginate again if we max the last page
+            if len(page["data"]["allInstrumentOverviews"]) < limit:
+                return
+
+            offset += limit
+
+    def scrape_versions(self, bill, row):
+        if row["IntroducedUrl"]:
+            bill.add_version_link(
+                "Introduced",
+                url=row["IntroducedUrl"],
+                media_type="application/pdf",
+            )
+        if row["EngrossedUrl"]:
+            bill.add_version_link(
+                "Engrossed",
+                url=row["EngrossedUrl"],
+                media_type="application/pdf",
+            )
+        if row["EnrolledUrl"]:
+            bill.add_version_link(
+                "Enrolled",
+                url=row["EnrolledUrl"],
+                media_type="application/pdf",
             )
 
-            bill.add_source("https://alison.legislature.state.al.us/bill-search")
+    def scrape_actions(self, bill, row):
+        if row["PrefiledDate"]:
+            action_date = datetime.datetime.strptime(row["PrefiledDate"], "%m/%d/%Y")
+            action_date = self.tz.localize(action_date)
+            bill.add_action(
+                chamber=self.chamber_map[row["Body"]],
+                description="Filed",
+                date=action_date,
+                classification="filing",
+            )
 
-            if row["IntroducedUrl"]:
-                bill.add_version_link(
-                    "Introduced", url=row["IntroducedUrl"], media_type="application/pdf"
-                )
+        headers = {
+            "Accept": "*/*",
+            "Accept-Language": "en-US,en;q=0.9",
+            "Authorization": "Bearer undefined",
+            "Content-Type": "application/json",
+            "Origin": "https://alison.legislature.state.al.us",
+            "Referer": "https://alison.legislature.state.al.us/",
+        }
 
-            # TODO: can this be removed in favor of pulling the
-            # action list via graphql?
-            if row["PrefiledDate"]:
-                action_date = datetime.datetime.strptime(
-                    row["PrefiledDate"], "%m/%d/%Y"
-                )
-                action_date = self.tz.localize(action_date)
-                bill.add_action(
-                    chamber=chamber,
-                    description="Filed",
-                    date=action_date,
-                    classification="filing",
-                )
+        # Can this be ANDED together with the other graphql query?
```

Comment:
@showerst I don't think so: all the properties on the `InstrumentOverviews` type (which is returned by the prior query request, `allInstrumentOverviews`) are scalar, there is no property that links to another object/interface. Same with what I assume is a related type, `InstrumentSummary`. Looking at the AL graphql schema I don't see much support for these kind of nested data that you normally expect in a graphql schema.